### PR TITLE
Actions for building and releasing manifests

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

Create and test 4 workflow for building and releasing.

1. workflow `build` which builds images in `amd64` and `arm64` arch respectively
2. workflow `release-charts` derives from original `release` workflow, which packs helm charts and release them and will be triggered by pushing tag "X.Y.Z"
3. workflow `release-images` builds images for `amd64` and `arm64` platforms and push them to specific docker registry. This workflow will be triggered by pushing tag "X.Y.Z"
4. workflow `release-distributions` builds and push images of `observer`, `obconfigserver`, `obproxy` and `obagent` with tag `[comp]-[version]`.

In workflow `release-images`, tag "X.Y.Z" will be used as tag of product images. And the following secrets and vars will be used in steps: 
* `secrets.DOCKER_USERNAME`, username to log in Docker Hub
* `secrets.DOCKER_PASSWORD`, password matching the given username
* `vars.DOCKER_PUSH_BASE`, should be `repo` in format `repo/image`, like `oceanbase` in oceanbase/ob-operator

### CODE REVIEW

File: .github/workflows/release-observer.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        run: echo "version=$(echo $tagName | grep -P '(\d*\.\d*\.\d*\.\d*-\d{18})' --only-matching)" >> $GITHUB_OUTPUT
+
+      - name: Build and push observer w/o cache
+        uses: docker/build-push-action@v2
+        with:
+          context: ./distribution/oceanbase
+          platforms: linux/amd64,linux/arm64
+          file: ./distribution/oceanbase/Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKER_USERNAME }}/oceanbase-cloud-native:${{ steps.set_version_vars.outputs.version }}
```
[the remark this answers is not in the document]
Comment (PowerfooI) [reply .github/workflows/release-observer.yml:41]: Maybe it should be a new global variable like `vars.DOCKER_REPO`

File: distribution/oceanbase/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,19 @@
 FROM golang:1.20.4 as builder
-ARG GOPROXY=https://goproxy.io,direct
```
[the remark this answers is not in the document]
Comment (PowerfooI) [reply distribution/oceanbase/Dockerfile:2]: In my machine, GOPROXY will the default `golang.org/xxx` if I keep this line.

**Comment by PowerfooI** [comment]
fix #107

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**